src/web/arr/trove/chart.arr

```diff
@@ -0,0 +1,1028 @@
+provide {
+  plot: plot,
+  plots: plots,
+
+  from-list: from-list,
+
+  image-plot: image-plot,
+  image-plots: image-plots,
+  draw-plot: draw-plot,
+  draw-plots: draw-plots,
+} end
+
+import global as G
+import base as B
+include lists
+import image-structs as I
+import image as IM
+import sets as S
+import chart-lib as P
+import either as E
+import string-dict as SD
+import valueskeleton as VS
+
+
+################################################################################
+# CONSTANTS
+################################################################################
+
+SHOW-LENGTH = 3
+FUNCTION-POINT-SIZE = 0.1
+DEFAULT-RANGE = {-10; 10}
+
+################################################################################
+# TYPE SYNONYMS
+################################################################################
+
+type PlottableFunction = (Number -> Number)
+type Posn = RawArray<Number>
+type TableIntern = RawArray<RawArray<Any>>
+
+################################################################################
+# HELPERS
+################################################################################
+
+fst = raw-array-get(_, 0)
+snd = raw-array-get(_, 1)
+posn = {(x :: Number, y :: Number, z :: String): [raw-array: x, y, z]}
```

Comment:
Cool to see a use of raw arrays for this. I think this is the first library that's really built this way with perf in mind. Does this make interop with the JS easier, too?

Reply:
There are two variants: `from-list` which we have now, and `from-table` which will be added after the new table interface is really finalized. Both variants construct the same `DataSeries`, and I hope that its `_output` will be table based.

I use `raw-array` as default because 1) that's what table is (underlying), so when we decide to implement fancy `_output`, it will be easier and 2) so that JS side needs not to worry about lists, though this is not really an issue.